Code/Framework/AzFramework/Platform/Common/WinAPI/AzFramework/IO/LocalFileIO_WinAPI.cpp

```diff
@@ -101,37 +79,30 @@ namespace AZ
 
         Result LocalFileIO::CreatePath(const char* filePath)
         {
-            char resolvedPath[AZ_MAX_PATH_LEN];
-            ResolvePath(filePath, resolvedPath, AZ_MAX_PATH_LEN);
+            FixedMaxPath resolvedPath;
+            ResolvePath(resolvedPath, filePath);
+            resolvedPath = resolvedPath.LexicallyNormal();
 
             // create all paths up to that directory.
             // its not an error if the path exists.
-            if ((Exists(resolvedPath)) && (!IsDirectory(resolvedPath)))
+            if (Exists(resolvedPath.c_str()) && !IsDirectory(resolvedPath.c_str()))
             {
                 return ResultCode::Error; // that path exists, but is not a directory.
             }
 
             // make directories from bottom to top.
-            AZStd::string buf;
-            size_t pathLength = strlen(resolvedPath);
-            buf.reserve(pathLength);
-            for (size_t pos = 0; pos < pathLength; ++pos)
+            FixedMaxPath directoryPath = resolvedPath.RootPath();
+            for (AZ::IO::PathView pathSegment : resolvedPath.RelativePath())
             {
-                if ((resolvedPath[pos] == '\\') || (resolvedPath[pos] == '/'))
+                directoryPath /= pathSegment;
+                CreateDirectoryA(directoryPath.c_str(), nullptr);
+                if (!IsDirectory(directoryPath.c_str()))
                 {
-                    if (pos > 0)
-                    {
-                        CreateDirectoryA(buf.c_str(), NULL);
-                        if (!IsDirectory(buf.c_str()))
-                        {
-                            return ResultCode::Error;
-                        }
-                    }
+                    return ResultCode::Error;
                 }
-                buf.push_back(resolvedPath[pos]);
             }
 
-            return SystemFile::CreateDir(buf.c_str()) ? ResultCode::Success : ResultCode::Error;
+            return SystemFile::CreateDir(directoryPath.c_str()) ? ResultCode::Success : ResultCode::Error;
```

Comment:
not sure why in old implementation it doesn't create directory for last path segment, but for new implementation do we need to create directory for root path first?

Reply:
The directory for the root path never needs to be created.
Because the root path is a path such as `D:/` on Windows or `/` on non-Windows platforms.

Those directory have to exist

Reply:
Also the final `SystemFile::CreateDir` call on Windows isn't needed, since  the new implementation will create the last path segment in the for loop.
I'll replace it with IsDirectory.